pkg/table/tables/index.go

```diff
@@ -546,7 +552,12 @@ func FetchDuplicatedHandle(ctx context.Context, key kv.Key, distinct bool,
 		return fetchDuplicatedHandleForTempIndexKey(ctx, key, distinct, txn, tableID, isCommon)
 	}
 	// The index key is not from temp index.
-	val, err := getKeyInTxn(ctx, txn, key)
+	var val []byte
+	if txn.IsPipelined() {
+		val, err = getKeyInPrefetchCache(ctx, txn, key)
```

Comment:
Are we sure this key is prefetched? If so we can add an assertion that it can be found from the cache. If not we should fallback to a txn.Get?